gtsfm/evaluation/metrics.py

```diff
@@ -310,7 +315,10 @@ def save_to_json(self, path: str) -> None:
         Args:
             path: path to json file.
         """
-        io.save_json_file(path, self.get_metrics_as_dict())
+        try:
+            io.save_json_file(path, self.get_metrics_as_dict())
```

Comment:
also curious, were some files not saving on some workers?

Reply:
I had problems with some metrics not being written to json because they could not be parsed. This was a way of not crashing the pipeline and knowing which metrics are causing the issue. I think its better than what we currently do, but also fine with not checking it in.